server/auth/auth.go

```diff
@@ -234,12 +234,18 @@ type httpClient interface {
 
 // IsUser returns true if the principal is a user.
 func IsUser(p knox.Principal) bool {
+	if mux, ok := p.(knox.PrincipalMux); ok {
+		p = mux.Default()
+	}
 	_, ok := p.(user)
```

Comment:
does it make more sense here to return true if any of the principals wrapped is a user? what happens if the order is different than expected?

Reply:
A) With the current config that isn't possible because if you pass user auth, you can't also be a service. Maybe worth changing in the future. The Spiffe Fallback Auth only works for machine auth right now.
B) The order is consistent in a given server. It is the order that you install the Auth Providers at startup. I was thinking that that order would provide a preference, and thus only checking the first one on these checks would make sense. But now you have me thinking also about the IsService check a few lines down.

I don't really have a preference between the two options. (Are any of them users, or is the first one a user.) Which do you prefer?

Reply:
I suppose its fine to go w/ the existing implementation, provided we document it (you can do this by adding some unit tests to prove that behavior).